src/plugins/data_views/common/data_views/ensure_default_data_view.ts

```diff
@@ -7,18 +7,20 @@
  */
 
 import { DataViewsContract } from './data_views';
-import { UiSettingsCommon } from '../types';
-
-export type EnsureDefaultDataView = () => Promise<unknown> | undefined;
+/**
+ * Checks whether a default data view is set and exists and defines
+ * one otherwise.
+ * @public
+ */
+export type EnsureDefaultDataView = () => Promise<unknown> | void;
 
-export const createEnsureDefaultDataView = (
-  uiSettings: UiSettingsCommon,
-  onRedirectNoDefaultView: () => Promise<unknown> | void
-) => {
-  /**
-   * Checks whether a default data view is set and exists and defines
-   * one otherwise.
-   */
+/**
+ * Checks whether a default data view is set and exists and defines
+ * one otherwise.
+ * @public
+ * @param onRedirectNoDefaultView - Callback to redirect to a new data view
+ */
+export const createEnsureDefaultDataView = (onRedirectNoDefaultView: EnsureDefaultDataView) => {
```

Comment:
Can we add a return type annotation here?

Reply:
Changed code so that `Promise<void> | void` is returned instead of `Promise<unknown> | void`